server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -206,31 +206,48 @@ def resource_name
-                                         :organization_id, :lifecycle_environment_id, :discovered_host_id,
-                                         :foreman_task_id, :openstack_deployment_id, :discovered_host_ids => [])
+      allowed = [
+        :name, :description, :deploy_rhev, :deploy_cfme,
+        :deploy_openstack, :deploy_openshift, :is_disconnected, :rhev_is_self_hosted,
+        :rhev_engine_admin_password, :rhev_data_center_name,
+        :rhev_cluster_name, :rhev_storage_name, :rhev_storage_type,
+        :rhev_storage_address, :rhev_cpu_type, :rhev_share_path,
+        :hosted_storage_name, :hosted_storage_address, :hosted_storage_path,
+        :cfme_install_loc, :rhev_root_password, :cfme_root_password,
+        :cfme_admin_password, :cfme_db_password, :foreman_task_uuid,
+        :upstream_consumer_uuid, :upstream_consumer_name, :rhev_export_domain_name,
+        :rhev_export_domain_address, :rhev_export_domain_path,
+        :rhev_local_storage_path, :rhev_gluster_node_name,
+        :rhev_gluster_node_address, :rhev_gluster_ssh_port,
+        :rhev_gluster_root_password, :host_naming_scheme, :has_content_error,
+        :custom_preprend_name, :enable_access_insights, :cfme_address, :cfme_hostname,
+        :openshift_install_loc, :openshift_number_master_nodes, :openshift_number_worker_nodes,
+        :openshift_storage_size, :openshift_username, :openshift_user_password,
+        :openshift_root_password, :openshift_master_vcpu, :openshift_master_ram,
+        :openshift_master_disk, :openshift_node_vcpu, :openshift_node_ram, :openshift_node_disk,
+        :openshift_available_vcpu, :openshift_available_ram, :openshift_available_disk,
+        :openshift_storage_type, :openshift_sample_helloworld, :openshift_storage_host,
+        :openshift_export_path, :openshift_subdomain_name, :cloudforms_vcpu,
+        :cloudforms_ram, :cloudforms_vm_disk_size, :cloudforms_db_disk_size,
+        :cdn_url, :manifest_file, :created_at, :updated_at, :rhev_engine_host_id,
+        :organization_id, :lifecycle_environment_id, :discovered_host_id,
+        :foreman_task_id, :openstack_deployment_id
+      ]
+
+      #############################################################
+      # Workaround for permitting the reset of discovered_hosts via accepting
+      # discovered_host_ids as an empty array. By default, if it's an empty array,
+      # strong params will filter the value so it does not impact an update.
+      # See discussion: https://github.com/rails/rails/issues/13766
+      #############################################################
+      if params[:deployment][:discovered_host_ids].nil?
+        allowed << :discovered_host_ids
+      else
+        allowed << { :discovered_host_ids => [] }
+      end
+      #############################################################
+
+      params.require(:deployment).permit(allowed)
```

Comment:
Looks like we're passing an array instead of spreading out the params
```ruby
params.require(:deployment).permit(allowed)
```
I was expecting:
```ruby
permit(*allowed)
```

It didn't cause you any issues?